thali/thalireplicationmanager.js

```diff
@@ -112,22 +117,32 @@ function networkChanged(status) {
   if (status.isAvailable && !this._isStarted) {
     this.start(this._deviceName, this._port, this._dbName);
   }
-}
+};
 
-/* synchronization */
+/**
+ * Synchronizes the peers from the peerAvailabiltyChanged event.
+ * @param {Object} peers Peers to sync which contain the peerIdentifier, peerAvailable and peerName.
+ */
+ThaliReplicationManager.prototype._syncPeers = function (peers) {
```

Comment:
It seems like this code is assuming that PEER_AVAILABILITY_CHANGED will always return all peers that are current visible? But in looking [here](https://github.com/thaliproject/Thali_CordovaPlugin/blob/story_0/doc/api/connectivity.md) it only says that the event will be called when a peer's availability changes. Which implies that if a peer's availability hasn't changed then it won't be included in the event. In which case we just overwrite this._peers with an incomplete list of who is available! That doesn't sound right. What am I missing?